src/AWSCredentials.jl

```
@@ -355,12 +358,15 @@ function ecs_instance_credentials()
 end
 
 """
-    env_var_credentials() -> Union{AWSCredential, Nothing}
+    env_var_credentials(explicit_profile::Bool=false) -> Union{AWSCredential, Nothing}
 
 Use AWS environmental variables (e.g. AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY, etc.)
 to create AWSCredentials.
 """
-function env_var_credentials()
+function env_var_credentials(explicit_profile::Bool=false)
+    # Skip using environmental variables when a profile has been explicitly set
+    explicit_profile && return nothing
```

Users could have an unintended affect here if they passed in a profile and have ENV VARs set. However I think moving towards matching the CLI is the proper step.

---

I personal consider this a bugfix as the old behavior did this:
```
❯ eval $(aws-vault export demo1 --format=export-env)

❯ aws sts get-caller-identity --query Arn --output text
arn:aws:sts:::assumed-role/demo1/1683754266745769000

❯ aws sts get-caller-identity --query Arn --output text --profile demo2
arn:aws:sts:::assumed-role/demo2/botocore-session-1683754083

❯ julia -e 'using AWS; @service STS; println(STS.get_caller_identity()["GetCallerIdentityResult"]["Arn"])'
arn:aws:sts:::assumed-role/demo1/1683754434179050000

❯ julia -e 'using AWS; @service STS; global_aws_config(; profile="s3utils-ci"); println(STS.get_caller_identity()["GetCallerIdentityResult"]["Arn"])'
arn:aws:sts:::assumed-role/demo1/1683754434179050000

❯ unset AWS_ACCESS_KEY_ID AWS_SECRET_ACCESS_KEY AWS_SESSION_TOKEN AWS_CREDENTIAL_EXPIRATION
```
Not in particular if you try to use a different profile AWS.jl would just silently ignore it.